src/cluster_legacy.c

```diff
@@ -5398,7 +5412,7 @@ static inline void removeAllNotOwnedShardChannelSubscriptions(void) {
 
 /* Set the specified node 'n' as primary for this node.
  * If this node is currently a primary, it is turned into a replica. */
-void clusterSetPrimary(clusterNode *n, int closeSlots) {
+void clusterSetPrimary(clusterNode *n, int closeSlots, int try_psync) {
```

Comment:
Nit - consider `full_sync_required`?

```suggestion
void clusterSetPrimary(clusterNode *n, int closeSlots, int full_sync_required) {
```